### PR TITLE
Add various readability/performance improvements

### PR DESCRIPTION
**Pull request by j8r**
* Refactor some if conditions
* Improve some String building
* Various little changes for code readability
* Formatting

### CODE REVIEW

**Comment by j8r** [comment]
Any thought @samueleaton ?

File: src/raze/methods.cr [hunk 1]
```diff
@@ -1,5 +1,6 @@
 HTTP_METHODS_OPTIONS = %w(get post put patch delete options)
 
+alias Caontext_types = HTTP::Server::Context | String | Int32 | Int64 | Bool | Nil
```
Comment (samueleaton) [inline src/raze/methods.cr]: typo `Caontext_types`

**Comment by j8r** [comment]
@samueleaton a bit worried about that not everything is speced, but existing ones pass.

**Comment by j8r** [comment]
@samueleaton have you the time to review this?

**Comment by j8r** [comment]
@samueleaton rebased :)

**Comment by j8r** [comment]
Closing, Kemal includes some of this improvements

**Pull request closed without merging**